doc/mgmt/SONiC_AliasFeature_HLD.md

```diff
@@ -0,0 +1,337 @@
+
+## 3.5 SAI
+
+## 3.6 User Interface
+
+### 3.6.1 Data Models
+Yang model used for alias mode handling  
+[sonic-device-metadata.yang](https://github.com/project-arlo/sonic-mgmt-framework/blob/master/models/yang/sonic/sonic-device-metadata.yang)  
+
+Supported yang objects and attributes are highlighted in green:
+```diff
+module: sonic-device-metadata
+    +--rw sonic-device-metadata
+       +--rw DEVICE_METADATA
+          +--rw DEVICE_METADATA_LIST* [name]
+             +--rw name                          string
++            +--rw aliasMode                     boolean
+```
+
+### 3.6.2 CLI
+
+#### 3.6.2.1 Configuration Commands
+
+#### Alias mode configuration
+
+`[no] alias enable`
+```
+sonic(config)# alias enable
+sonic(config)# no alias enable
+```
+#### Config commands when alias mode is enabled
+`Eth 1/2`, `Eth1/2`, and `e1/2` options are supported to get into alias interface config mode.  The parser converts all to `Eth1/2`.
+
+```
+sonic(config)# interface e1/2
+sonic(conf-if-Eth1/2)#
+```
+```
+sonic(conf-if-Eth1/2)# ip address 2.2.2.2/24
+```
+
+#### 3.6.2.2 Show Commands
+
+#### Display alias mode  
+
+`show alias`
+```
+sonic(config)# alias enable
+sonic# show alias
+Alias mode is enabled
+```
+```
+sonic(config)# no alias enable
+sonic# show alias
+Alias mode is disabled
+```
+#### Show commands when alias mode is enabled
+```
+show vlan
+Q: A - Access (Untagged), T - Tagged
+NUM        Status      Q Ports
+2          Inactive    A  Eth1/3
+                       T  Eth1/5/2
+                       T  Eth1/10
+```
+
+
+
+
+#### 3.6.2.3 Debug Commands
+
+#### 3.6.2.4 IS-CLI Compliance
+
+The following table maps SONiC CLI commands to corresponding IS-CLI commands. The compliance column identifies how the command comply to the IS-CLI syntax:
+
+- **IS-CLI drop-in replace**  \u2013 meaning that it follows exactly the format of a pre-existing IS-CLI command.
+- **IS-CLI-like**  \u2013 meaning that the exact format of the IS-CLI command could not be followed, but the command is similar to other commands for IS-CLI (e.g. IS-CLI may not offer the exact option, but the command can be positioned is a similar manner as others for the related feature).
+- **SONiC** - meaning that no IS-CLI-like command could be found, so the command is derived specifically for SONiC.
+
+| CLI Command | Compliance | IS-CLI Command (if applicable) | Link to the web site identifying the IS-CLI command (if applicable) |
+| ----------- | ---------- | ------------------------------ | ------------------------------------------------------------ |
+|             |            |                                |                                                              |
+
+### 3.6.3 REST API Support
+
+#### 3.6.3.1
+
+##### Following REST operations will be supported
+
+**PATCH, PUT, DELETE and GET**
+
+- `​/sonic-device-metadata:sonic-device-metadata​/DEVICE_METADATA​/DEVICE_METADATA_LIST={name}​/aliasMode`
+
+
+# 4 Flow Diagrams
+
+# 5 Error Handling
+

```

Comment:
What happens if the user (CLI/script, REST, gNMI) refers to the wrong naming mode? Pls show the error message returned.